packages/front-end/components/Experiment/PValueGuardrailResults.tsx

```diff
@@ -0,0 +1,176 @@
+import { MetricInterface } from "back-end/types/metric";
+import { ExperimentReportVariation } from "back-end/types/report";
+import usePValueThreshold from "@/hooks/usePValueThreshold";
+import Tooltip from "../Tooltip/Tooltip";
+import MetricTooltipBody from "../Metrics/MetricTooltipBody";
+import MetricValueColumn from "./MetricValueColumn";
+
+type PValueGuardrailResult = {
+  stats: SnapshotMetric;
+  expectedDirection: boolean;
+  statSig: boolean;
+  users: number;
+  name: string;
+  hasEnoughData: boolean;
+};
+
+function hasEnoughData(value1: number, value2: number): boolean {
+  // TODO Is this right for pvalue?
+  return Math.max(value1, value2) >= 80 && Math.min(value1, value2) >= 0.0001;
+}
+
+const HeaderResult: FC<{
+  metric: MetricInterface;
+  results: PValueGuardrailResult[];
+}> = ({ metric, results }) => {
+  const anyInsufficientData = results.some((r) => !r.hasEnoughData);
+  const significantNegativeDirection = results.some(
+    (r) => !r.expectedDirection && r.statSig
+  );
+  const anyNegativeDirection = results.some((r) => !r.expectedDirection);
+  const allSignificantPositiveDirection = results.every(
+    (r) => r.expectedDirection && r.statSig
+  );
+
+  let status = "secondary";
+
+  if (anyInsufficientData) {
+    status = "secondary";
+  } else if (significantNegativeDirection) {
+    status = "danger";
+  } else if (anyNegativeDirection) {
+    status = "warning";
+  } else if (allSignificantPositiveDirection) {
+    status = "success";
+  }
+
+  return (
+    <div
+      className={clsx(
+        "d-flex align-items-center guardrail alert m-0",
+        `alert-${status}`
+      )}
+    >
+      {status === "success" && <FaCheckCircle className="mr-1" />}
+      {status === "warning" && <FaExclamationTriangle className="mr-1" />}
+      {status === "danger" && <FaExclamation className="mr-1" />}
+      {status === "secondary" && <FaQuestionCircle className="mr-1" />}
+      <Tooltip body={<MetricTooltipBody metric={metric} />} tipPosition="right">
+        <Link href={`/metric/${metric.id}`}>
+          <a className="text-dark font-weight-bold">{metric.name}</a>
+        </Link>
+      </Tooltip>
+    </div>
+  );
+};
+
+const PValueGuardrailResults: FC<{
+  data: SnapshotVariation[];
+  variations: ExperimentReportVariation[];
+  metric: MetricInterface;
+}> = ({ data, variations, metric }) => {
+  const pValueThreshold = usePValueThreshold();
+
+  const results: PValueGuardrailResult[] = useMemo(() => {
+    return variations.map((v, i) => {
+      const stats = data[i]?.metrics?.[metric.id];
+      const expectedDirection = metric.inverse
+        ? stats.expected < 0
+        : stats.expected > 0;
+      const statSig = stats.pValue < pValueThreshold;
+      const users = data[i].users;
+      const name = v.name;
+      return {
+        stats,
+        expectedDirection,
+        statSig,
+        users,
+        name,
+        hasEnoughData: hasEnoughData(
+          stats.pValue,
```

Comment:
So here you need to just pass the baseline value and the variation value. The pvalue itself isn't used in determining `hasEnoughData`

Reply:
Got it